tests/playwright/Report.md

```diff
@@ -0,0 +1,213 @@
+
+[Playwright](https://github.com/microsoft/playwright) is a Node.js library to automate Chromium, Firefox, and WebKit with a single API. Playwright is known for the following capability;
+
+* Test across all modern browsers: Single API to automate Chromium, Firefox, and WebKit.
+* Use in your preferred language: Use the Playwright API in JavaScript & TypeScript, Python, C# and, Java.
+* Automate without trade-offs: Capable automation for single-page apps that rely on the modern web platform.
+
+The above points are just a few capabilities. To explore more on Playwright capability, you can check out the [Playwright](https://playwright.dev/) documentation and my [MediaWiki Core fork](https://github.com/hariclerry/mediawiki/tree/master/tests/playwright).
+
+While working with Playwright for the past two months, it has come across as easy to install and setup. The package takes care of installing all the browsers(Chromium, Firefox, and WebKit). Capturing Screenshots is an out-of-the-box experience. However, video recording requires separate installation of [playwright-video and ffmpeg](https://playwright.tech/blog/record-your-browser-tests-with-playwright) but they all blend in with Playwright seamlessly.
+
+Below are some of the benefits I have experienced and seen while using Playwright;
+
+* Console debug option which is very useful in an event when tests are failing with no clear error messages.
+* Scenarios that span multiple pages, domains, and iframes
+* Auto-wait for elements to be ready before executing actions (like click, fill)
+* Intercept network activity for stubbing and mocking network requests
+* Seamless integration with Jest.
+
+#### Sample Playwright Page screenshot code
+
+This code snippet navigates to wikipedia.org in WebKit and saves the screenshot.
+
+```js
+const playwright = require("playwright");
+
+(async () => {
+ const browser = await webkit.launch();
+ const context = await browser.newContext();
+ const page = await context.newPage();
+ await page.goto("https://www.wikipedia.org/");
+ await page.screenshot({ path: "wikipedia-home-page.png" });
+ await browser.close();
+})();
+```
+
+#### Playwright and Jest
+
+[Jest](https://jestjs.io/) is a delightful JavaScript Testing Framework with a focus on simplicity. Playwright can be integrated into a project in two ways and one of them is by using the [jest-playwright](https://github.com/playwright-community/jest-playwright) which has rich features like:
+
+* Multi-browser and device (like iPhones with given screen sizes) support
+* jest-dev-server integration which can start your webserver like create-react-app before running the tests
+* expect-playwright integration which provides common expect helper functions
+
+jest-playwright is added to the jest configuration as a preset which makes all the features availbale to be used at your disposal. It was inspired by [jest-puppeteer](https://github.com/smooth-code/jest-puppeteer/blob/master/README.md#start-a-server).
```

Comment:
Please install Code Spell Checker to help you find typos in the future.

https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker

```suggestion
jest-playwright is added to the jest configuration as a preset which makes all the features available to be used at your disposal. It was inspired by [jest-puppeteer](https://github.com/smooth-code/jest-puppeteer/blob/master/README.md#start-a-server).
```